openssl-sys/src/asn1.rs

```diff
@@ -61,14 +61,8 @@ extern "C" {
     pub fn ASN1_TIME_set_string_X509(s: *mut ASN1_TIME, str: *const c_char) -> c_int;
 }
 
-cfg_if! {
-    if #[cfg(any(ossl110, libressl280))] {
-        extern "C" {
-            pub fn ASN1_STRING_to_UTF8(out: *mut *mut c_uchar, s: *const ASN1_STRING) -> c_int;
-        }
-    } else {
-        extern "C" {
-            pub fn ASN1_STRING_to_UTF8(out: *mut *mut c_uchar, s: *mut ASN1_STRING) -> c_int;
-        }
+const_ptr_api! {
+    extern "C" {
+        pub fn ASN1_STRING_to_UTF8(out: *mut *mut c_uchar, s: #[const_ptr_if(any(ossl110, libressl280))] ASN1_STRING) -> c_int;
```

Comment:
Can the macro be changed to only apply to the individual extern definition rather than wrapping the whole extern "C" block?

Reply:
Syntactically speaking I think the macro can do that, but rustc doesn't like it; [playground example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=677f536409246611f7feb259db9d56fd) fails with `error[E0428]: the name `ASN1_STRING_to_UTF8` is defined multiple times`; but when looking at the generated code through `Tools` -> `Expand macros` it seems to be doing the right thing.

I'm pretty sure that is the reason why the `cfg_if!` blocks have internal `extern "C" {...}` blocks too; probably the same limitation. And even if `rustc` got fixed and you raise the MSRV, ctest will still fail, as it uses an old fork if a rustc internal crates to do parsing and macro expansion.

Passing `extern "C" { ... }` can be removed of course, but it can't be part of another `extern "C" { ... }` block and needs to emit `extern "C" { ... }` for the fn items.

Reply:
Ah bummer. This seems good then!